src/Npgsql/MultiplexingConnectorPool.cs

```diff
@@ -266,12 +266,12 @@ bool WriteCommand(NpgsqlConnector connector, NpgsqlCommand command, ref Multiple
                     {
                         // If this statement isn't prepared, see if it gets implicitly prepared.
                         // Note that this may return null (not enough usages for automatic preparation).
-                        if (!statement.IsPrepared)
-                            statement.PreparedStatement = connector.PreparedStatementManager.TryGetAutoPrepared(statement);
-                        if (statement.PreparedStatement is PreparedStatement pStatement)
+                        if (!statement.IsPrepared(out var pStatement))
+                            pStatement = statement.PreparedStatement = connector.PreparedStatementManager.TryGetAutoPrepared(statement);
```

Comment:
Note that here `statement.PreparedStatement` is accessed twice (once from `statement.IsPrepared` and once while setting `pStatement`). One way to mitigate this is by setting `pStatement` before `statement.IsPrepared`, but it does change the existing behavior.